iOS/Flows/Conversation/ConversationListCollectionViewLayout.swift

```diff
@@ -8,29 +8,39 @@
 
 import Foundation
 
+protocol ConversationListCollectionViewLayoutDelegate: AnyObject {
+    func conversationListCollectionViewLayout(_ layout: ConversationListCollectionViewLayout,
+                                              cidFor indexPath: IndexPath) -> ConversationId?
+    func conversationListCollectionViewLayout(_ layout: ConversationListCollectionViewLayout,
+                                              didUpdateCentered cid: ConversationId?)
+}
+
 /// A custom layout class for conversation collection views. It lays out its contents in a single horizontal row.
-/// If the conversation collectionview's semantic content attribute is rightToLeft, the first cell will be on the far right.
 class ConversationListCollectionViewLayout: UICollectionViewFlowLayout {
 
+    weak var delegate: ConversationListCollectionViewLayoutDelegate?
+    private var previousCenteredCID: ConversationId?
+
     override init() {
         super.init()
-
         self.scrollDirection = .horizontal
     }
 
     required init?(coder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
+        super.init(coder: coder)
+        self.scrollDirection = .horizontal
     }
 
-    // IMPORTANT: Returning true allows us to layout the cells right to left.
-    override var flipsHorizontallyInOppositeLayoutDirection: Bool {
-        return true
+    override func shouldInvalidateLayout(forBoundsChange newBounds: CGRect) -> Bool {
+        self.sendCenterUpdateEventIfNeeded(withContentOffset: newBounds.origin)
+
+        return super.shouldInvalidateLayout(forBoundsChange: newBounds)
```

Comment:
Whenever the bounds change, we check to see if we need to update the centered cell. Duplicate updates are ignored.